lib/containers/engine.pm

```diff
@@ -0,0 +1,396 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020-2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Abstraction layer to operate docker and podman containers through same interfaces
+# Maintainer: qac team <qa-c@suse.de>
+
+package containers::engine;
+use Mojo::Base -base;
+use testapi;
+use Test::Assert 'assert_equals';
+use containers::utils qw(registry_url);
+use utils qw(systemctl file_content_replace);
+
+has runtime => undef;
+
+sub _engine_assert_script_run {
+    my ($self, $cmd, @args) = @_;
+    assert_script_run($self->runtime . " " . $cmd, @args);
+}
+
+sub _engine_script_run {
+    my ($self, $cmd, @args) = @_;
+    return script_run($self->runtime . " " . $cmd, @args);
+}
+
+sub _engine_script_output {
+    my ($self, $cmd, @args) = @_;
+    return script_output($self->runtime . " " . $cmd, @args);
+}
+
+=head2 create_container($image, $name, [$cmd])
+
+Creates a container.
+C<image> the name of the image we create the container from.
+C<name> is the given name of the created container.
+C<cmd> can be anything you want to run in the container, similar to C<run>. for 
+instance: C<docker create -it tumbleweed bash>
+
+=cut
+sub create_container {
+    my ($self) = shift;
+    my %args = (
+        image => '',
+        name  => '',
+        cmd   => '',
+        @_
+    );
+    die('Must provide an image') unless ($args{image});
+    die('Must provide an name')  unless ($args{name});
+    $self->_engine_assert_script_run("container create --name $args{name} $args{image} $args{cmd}", 300);
+}
+
+=head2 start_container($image_name)
+
+Starts container named C<image_name>.
+
+=cut
+sub start_container {
+    my ($self, $image_name) = @_;
+    die 'wrong number of arguments' if @_ < 2;
```

Comment:
Don't need this. If you specify the args like this instead of %args, if will fail if you make a call like `start_container()`. It will complain that it requires 1 argument.

So, 2 options:
1) you keep it like this and remove the unnecessary `die`
2) you switch to %args and keep the `die unless ($args{image_name});`

Any of the 2 are ok for me.

Reply:
*this chage is ready for review* going with (1)